qcodes/plots/qcmatplotlib.py

```diff
@@ -288,6 +295,57 @@ def _draw_plot(self, ax, y, x=None, fmt=None, subplot=1,
         line, = ax.plot(*args, **kwargs)
         return line
 
+    @staticmethod
+    def _make_args_for_pcolormesh(args_masked, x, y):
+        if x is not None and y is not None:
+            # If x and y are provided, modify the arrays such that they
+            # correspond to grid corners instead of grid centers.
+            # This is to ensure that pcolormesh centers correctly and
+            # does not ignore edge points.
+            args = []
+            for k, arr in enumerate(args_masked[:-1]):
+                # If a two-dimensional array is provided, only consider the
+                # first row/column, depending on the axis
+                if arr.ndim > 1:
+                    arr = arr[0] if k == 0 else arr[:, 0]
+                # first extrapolate to fill any empty values Matplotlib 2.2 no
+                # longer support nans in x and y for pcolormesh.
+                if np.ma.is_masked(arr[1]):
+                    step_size = 1.
+                    # Only the first element is not nan. We have to guess the
+                    # step size
+                else:
+                    # the average stepsize is our best guess
+                    step_size = np.ma.average(np.ma.diff(arr))
+
+                last_good_value = arr[np.logical_not(arr.mask)][-1]
+                extrapolation_start = last_good_value+step_size
+                n_invalid = np.sum(arr.mask)
+                extrapolation_stop = extrapolation_start+step_size*(n_invalid-1)
+                # numpy (1.14) has a deprecation warning related to shared
+                # masks
+                # lets silence this by making sure that this is not shared
+                # before
```

Comment:
PEP8 is nice and all but IMHO this indentation is worse than too long lines 😛

Reply:
Yeah, I'm still getting used to VSCode. That would just have been `meta+q` in Emacs... sigh...